packages/site/cms-site/src/sitePreview/pagesRouter/sitePreviewApiHandler.ts

```diff
@@ -0,0 +1,51 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { Scope, SitePreviewParams } from "../SitePreviewUtils";
+
+type GraphQLClient = {
+    setHeader(key: string, value: string): unknown;
+    request<Response, Variables>(document: string, variables?: Variables): Promise<Response>;
+};
+
+async function sitePreviewApiHandler(req: NextApiRequest, res: NextApiResponse, graphQLClient: GraphQLClient) {
+    const params = req.query;
+    const settingsParam = params.settings;
+    const scopeParam = params.scope;
+
+    if (typeof settingsParam !== "string" || typeof scopeParam !== "string") {
+        throw new Error("Missing settings or scope parameter");
+    }
+
+    if (Array.isArray(params.path)) {
+        res.status(400).json({ error: "Parameter 'path' can't be an array" });
+        return;
+    }
+
+    const previewData = JSON.parse(settingsParam);
+    const scope = JSON.parse(scopeParam);
+
+    graphQLClient.setHeader("authorization", req.headers?.authorization ?? "");
+
+    const { currentUser } = await graphQLClient.request<{ currentUser: { permissionsForScope: string[] } }, { scope: Scope }>(
+        `
+            query CurrentUserPermissionsForScope($scope: JSONObject!) {
+                currentUser {
+                    permissionsForScope(scope: $scope)
+                }
+            }
+        `,
+        { scope },
+    );
+    if (!currentUser.permissionsForScope.includes("pageTree")) {
+        res.status(403).json({ error: "Preview is not allowed" });
+        return;
+    }
+
+    const data: SitePreviewParams = { scope, previewData };
+
+    res.setDraftMode({ enable: true });
+    res.setPreviewData(data);
```

Comment:
do you need both?

Reply:
What do you mean? Both function calls, or both Preview and Draft Mode?

Reply:
both Preview and Draft Mode, can't you just use preview mode, like we had before comet v7?

Reply:
Changed to preview mode only: 78f6ccf933fa8b53c7c6dd9b9944bd234c4c4520

Reply:
Do we need to encode the preview params as we do for App Router? @fraxachun what do you think?

Reply:
No, that is secured by nextjs, iirc

Reply:
> However, if it still works in app router, why not. The advantage is that we don't have to store the preview data by our own.

We only use preview mode for the pages router. App router still uses draft mode.

> Maybe read it in the middleware?

I don't think this would help with the migration to v7. Also, how would I then pass the info to getStaticProps?